main.py

```diff
@@ -67,16 +67,40 @@ def delete(self, dpid=None):
         """
         return self._send_flow_mods_from_request(dpid, "delete")
 
-    def _send_flow_mods_from_request(self, dpid, command):
-        flows_dict = request.get_json()
+    def _get_enabled_switch(self, dpid=None):
+        """Get enabled switch specified by dpid.
 
-        if dpid is None:
-            switches = self.controller.switches.values()
+        If no dpid is specified, return all switches enabled.
+
+        Returns:
+            list: A list with all switches enabled.
+
+        Raises:
+           Exception: Exception message when switches is not found.
+        """
+        switches = None
+        if dpid:
+            switch = self.controller.get_switch_by_dpid(dpid)
+            if switch is None:
+                raise Exception('dpid not found.')
+            elif switch.enabled is False:
+                raise Exception('switch is disabled.')
+            else:
+                switches = [switch]
         else:
-            switches = [self.controller.get_switch_by_dpid(dpid)]
+            all_switches = self.controller.switches.values()
+            switches = [switch for switch in all_switches if switch.enabled]
+            if len(switches) == 0:
+               raise Exception('all switches are disabled.')
+        return switches
```

Comment:
the method name is `get_enabled_switch` and you are return a list. 

Rename to plural.